flake.nix

```diff
@@ -0,0 +1,52 @@
+{
+  description = "Rustlings devShell.";
+
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, flake-compat, flake-utils, nixpkgs }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        rustlings = pkgs.rustPlatform.buildRustPackage {
+          name = "rustlings";
+          version = "5.1.0";
+
+          src = with pkgs.lib; cleanSourceWith {
+            src = self;
+            # a function that returns a bool determining if the path should be included in the cleaned source
+            filter = path: type:
+              let
+                # filename
+                baseName = builtins.baseNameOf (toString path);
+                # path from root directory
+                path' = builtins.replaceStrings [ "${self}/" ] [ "" ] path;
+                # checks if path is in the directory
+                inDirectory = directory: hasPrefix directory path';
+              in
+              inDirectory "src" ||
+              inDirectory "tests" ||
+              hasPrefix "Cargo" baseName ||
+              baseName == "default_out.txt" ||
```

Comment:
What is this path? I don't see it mentioned anywhere in the source.

Reply:
Sorry, but what path do you mean?